src/Security/Authorization/Core/src/AuthorizationPolicy.cs

```diff
@@ -176,17 +176,6 @@ public static async Task<AuthorizationPolicy> CombineAsync(IAuthorizationPolicyP
                 }
             }
 
-            var requiredPolicy = await policyProvider.GetRequiredPolicyAsync();
```

Comment:
If we remove this, we should modify `AuthorizationMiddleware` to no-op if no auth metadata is available for the endpoint. 

See https://github.com/aspnet/AspNetCore/blob/master/src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs#L53:

```C#
var authorizeData = endpoint?.Metadata.GetOrderedMetadata<IAuthorizeData>();
if (authorizeData == null)
{
   return _next(context);
}
```

cc @JamesNK \ @rynowak  to confirm this behavior.

Reply:
Could someone have a policy provider - https://github.com/aspnet/AspNetCore/blob/1cd544042d6fba73f844b5e6d5f10d9e7dfa0367/src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs#L54 - that still returns a policy even if there is no authz metadata?

Reply:
https://github.com/aspnet/AspNetCore/blob/1cd544042d6fba73f844b5e6d5f10d9e7dfa0367/src/Security/Authorization/Core/src/AuthorizationPolicy.cs

It looks like that after this change the policy provider would never be call if there is policy metadata. No-oping would be good. We could even skip the async state machine in the AuthZ middleware if there is no metadata.

Reply:
> - that still returns a policy even if there is no authz metadata

As you noted, CombineAsync does not call the policy provider in the event there's no auth metadata.

Reply:
Added a no-op check if there's no authorize data to the middleware
```C#
            if (authorizeData.Length == 0) {
                await _next(context);
                return;
            }
```